pkg/firewall/firewall.go

```diff
@@ -0,0 +1,175 @@
+package firewall
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ipfs/go-log"
+
+	coreChain "github.com/keep-network/keep-core/pkg/chain"
+	coreFirewall "github.com/keep-network/keep-core/pkg/firewall"
+	coreNet "github.com/keep-network/keep-core/pkg/net"
+	coreKey "github.com/keep-network/keep-core/pkg/net/key"
+
+	eth "github.com/keep-network/keep-ecdsa/pkg/chain"
+)
+
+var logger = log.Logger("keep-firewall")
+
+// KeepCacheLifetime is the time the cache maintains the list of active keep
+// members. We use the cache to minimize calls to Ethereum client.
+const KeepCacheLifetime = 24 * time.Hour
```

Comment:
When we were discussing this PR we were thinking about increasing the cache lifetime to 7 days. I just want to confirm that we didn't miss an update here and you finally decided to go with 1 day.

Reply:
Done in https://github.com/keep-network/keep-ecdsa/pull/406/commits/263b1773e379e0d81245dde9f4ef1c4f7caaf2a7